app/src/main/java/com/springsamurais/toyswap/service/RetrofitInstance.kt

```diff
@@ -4,20 +4,15 @@ import retrofit2.Retrofit
 import retrofit2.converter.gson.GsonConverterFactory
 import retrofit2.create
 
-class RetrofitInstance {
-
-    private var retrofit: Retrofit? = null
+object RetrofitInstance {
 
     private val baseUrl: String = "http://10.0.2.2:8080/api/v1/"
 
-    fun getService(): ListingAPIService {
-        if (retrofit == null) {
-            retrofit = Retrofit.Builder()
-                .baseUrl(baseUrl)
-                .addConverterFactory(GsonConverterFactory.create())
-                .build()
-        }
-        return retrofit!!.create(ListingAPIService::class.java)
+    val instance: APIService by lazy {
+        val retrofit = Retrofit.Builder()
+            .baseUrl(baseUrl)
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+        retrofit.create(APIService::class.java)
     }
-
 }
```

Comment:
Hi, are we not adhering to the singleton pattern? I guess, if we make call during request we will make more instances of the class